pkg/api/validate/dynamic/encryptionathost_test.go

```diff
@@ -19,30 +19,85 @@ import (
 )
 
 func TestValidateEncryptionAtHost(t *testing.T) {
+
+	EncryptionAtHostRegistered := func() *api.SubscriptionProperties {
+		return &api.SubscriptionProperties{
+			RegisteredFeatures: []api.RegisteredFeatureProfile{
+				{
+					Name:  "Microsoft.Compute/EncryptionAtHost",
+					State: "Registered",
+				},
+			},
+		}
+	}
+
+	EncryptionAtHostEnabled := func(MasterVMSize api.VMSize, WorkerVMSize api.VMSize) *api.OpenShiftCluster {
+		return &api.OpenShiftCluster{
+			Properties: api.OpenShiftClusterProperties{
+				MasterProfile: api.MasterProfile{
+					EncryptionAtHost: api.EncryptionAtHostEnabled,
+					VMSize:           MasterVMSize,
+				},
+				WorkerProfiles: []api.WorkerProfile{{
+					EncryptionAtHost: api.EncryptionAtHostEnabled,
+					VMSize:           WorkerVMSize,
+				}},
+			},
+		}
+	}
+
+	EncryptionAtHostEnabledOrDisabled := func(MasterEncryptionAtHost api.EncryptionAtHost, WorkerEncryptionAtHost api.EncryptionAtHost) *api.OpenShiftCluster {
```

Comment:
I like the refactor here to keep the actual test cases focused on their intent, but I think the function name and syntax here is a little confusing and ends up obscuring some important information when looking at the test cases themselves, e.g. it's unclear from just reading `EncryptionAtHostEnabledOrDisabled(api.EncryptionAtHostEnabled, api.EncryptionAtHostDisabled)` what that test case is trying to set up. 

I wonder if it might be cleaner to establish static master/workerprofile definitions, and have each test case assemble the cluster definition by composing those static members (e.g. `func(masterProfile api.MasterProfile, workerProfile api.WorkerProfile)`), or to just make a single, four-arg function that handles both cases.